@here/harp-mapview/lib/text/Placement.ts

```diff
@@ -617,19 +649,31 @@ function placePointLabelAtAnchor(
         tmpMeasurementParams.letterCaseArray = label.glyphCaseArray!;
         // Compute label bounds according to layout settings.
         textCanvas.measureText(label.glyphs!, label.bounds, tmpMeasurementParams);
+        // TODO: Make the margin configurable
+        label.bounds.expandByVector(pointLabelMargin);
     }
 
     // Compute text offset from the anchor point
     const textOffset = computePointTextOffset(label, placement, scale, env, tmpTextOffset);
     textOffset.add(screenPosition);
     tmpBox.copy(label.bounds!);
-    tmpBox.min.multiplyScalar(scale);
-    tmpBox.max.multiplyScalar(scale);
-    // Add margin after scaling, this ensures the margin is consistent across all
-    // labels - regardless of distance scaling (or any other) factor.
-    // TODO: Make the margin configurable
-    tmpBox.expandByVector(pointLabelMargin);
+    tmpBox.getCenter(tmpCenter);
+    tmpBox.getSize(tmpSize);
+    if (measureText) {
+        // New label padding/margin is applied only for the first time it is placed.
+        // This margin compensates the changes of label bounds due to distance scaling and
+        // it is designed to cover only size changes due to camera panning (not tilt).
+        // This ways label "reserves" enough space so it may "grow" without affecting other
+        // labels, no collisions occurs and thus limits labels flickering on pan.
+        const imprecisionFactor = 1.1;
+        // Account for imprecisions during calculus - increase the scale by 10%.
+        tmpSize.multiplyScalar(maxPlanarScale * imprecisionFactor);
```

Comment:
Please note, that we could also simply use some Theme defined / hard-coded margin as above - see:
```
label.bounds.expandByVector(pointLabelMargin);
```
Saying that the margin above is `persistentLabelMargin` and the new one is `newLabelMaring`.
Although the effect will be the same the only difference with this PR solution is that we do not have any assumed values but calculate scaling on the fly... of course with some CPU cost.
I am leaving the decision to you dear reviewers :).